pkg/odo/cli/component/create.go

```diff
@@ -192,25 +208,90 @@ func (co *CreateOptions) setResourceLimits() {
+	if len(args) == 0 || !cmd.HasFlags() {
+		co.interactive = true
+	}
+
 	co.Context = genericclioptions.NewContextCreatingAppIfNeeded(cmd)
 	co.CreateArgs.ApplicationName = co.Context.Application
 
-	co.CreateArgs.Wait = co.wait
+	if co.interactive {
+		client := co.Client
 
-	err = co.setCmpSourceAttrs()
-	if err != nil {
-		return err
-	}
+		componentTypeCandidates, err := catalog.List(client)
+		if err != nil {
+			return err
+		}
+		componentTypeCandidates = catalogutil.FilterHiddenComponents(componentTypeCandidates)
+		selectedComponentType := ui.SelectComponentType(componentTypeCandidates)
+		selectedImageTag := ui.SelectImageTagInteractively(componentTypeCandidates, selectedComponentType)
+		co.CreateArgs.ImageName = selectedComponentType + ":" + selectedImageTag
+
+		selectedSourceType := ui.SelectSourceType([]occlient.CreateType{occlient.LOCAL, occlient.GIT, occlient.BINARY})
+		co.CreateArgs.SourceType = selectedSourceType
+		selectedSourcePath := ""
+		if selectedSourceType == occlient.LOCAL {
+			currentDirectory, err := util.GetAbsPath("./")
+			if err != nil {
+				return err
+			}
+			selectedSourcePath = ui.EnterInputTypePath("local", currentDirectory)
+		} else if selectedSourceType == occlient.BINARY {
+			selectedSourcePath = ui.EnterInputTypePath("binary", "")
+		} else if selectedSourceType == occlient.GIT {
+			var selectedGitRef string
+			selectedSourcePath, selectedGitRef = ui.EnterGitInfo()
+			co.CreateArgs.SourceRef = selectedGitRef
+		}
+		co.CreateArgs.SourcePath = selectedSourcePath
 
-	err = co.setCmpName(args)
-	if err != nil {
-		return err
-	}
+		defaultComponentName, err := createDefaultComponentName(co.Context, selectedComponentType, selectedSourceType, selectedSourcePath)
+		if err != nil {
+			return err
+		}
+		co.CreateArgs.Name = ui.EnterComponentName(defaultComponentName)
+
+		advancedOptionsNeeded := ui.SelectYesNo("Do you wish to set advanced options")
+		if advancedOptionsNeeded {
+			co.CreateArgs.Ports = ui.EnterPorts()
+			co.CreateArgs.EnvVars = ui.EnterEnvVars()
+
+			resourceLimitsNeeded := ui.SelectYesNo("Do you wish to set resource limits")
+			if resourceLimitsNeeded {
+				memMin := ui.EnterMemory("minimum", "50Mi")
+				memMax := ui.EnterMemory("maximum", "512Mi")
+				cpuMin := ui.EnterCPU("minimum", "100m")
+				cpuMax := ui.EnterCPU("maximum", "1")
+
+				resourceQuantity := []util.ResourceRequirementInfo{}
+				memoryQuantity := util.FetchResourceQuantity(corev1.ResourceMemory, memMin, memMax, "")
+				if memoryQuantity != nil {
+					resourceQuantity = append(resourceQuantity, *memoryQuantity)
+				}
+				cpuQuantity := util.FetchResourceQuantity(corev1.ResourceCPU, cpuMin, cpuMax, "")
+				if cpuQuantity != nil {
+					resourceQuantity = append(resourceQuantity, *cpuQuantity)
+				}
+				co.CreateArgs.Resources = resourceQuantity
+			}
 
-	co.setResourceLimits()
+			co.CreateArgs.Wait = ui.SelectYesNo("Would you wish to wait until the component is fully ready after after creation")
```

Comment:
I think you're missing an assignment to `co.wait` as well here since I still get the `this might take a few moments` message even if I asked to wait.

Reply:
Indeed yes

Reply:
Fixed